src/main/java/edu/ucsb/cs56/ucsb_courses_search/controller/ScheduleController.java

```diff
@@ -95,19 +97,50 @@ public String add( @PathVariable("scheduleid") Long scheduleid,
         primary.setQuarter(lecture_quarter);
         primary.setScheduleid(scheduleid);
         logger.info("primary = " + primary); 
-        scheduleItemRepository.save(primary);
+        // scheduleItemRepository.save(primary);
 
         return "redirect:/schedule";
     }
 
     @PostMapping("/schedule/create")
-    public String add_schedule(String sname, Model model, OAuth2AuthenticationToken token) {
+    public String add_schedule(Model model, OAuth2AuthenticationToken token) {
         Schedule newschedule = new Schedule();
         String uid = token.getPrincipal().getAttributes().get("sub").toString();
         newschedule.setUid(uid);
-        newschedule.setSchedulename(sname);
+        newschedule.setSchedulename("New Schedule");
         scheduleRepository.save(newschedule);
         return "redirect:/schedule";
     }
 
+    @GetMapping("/schedule/{scheduleid}")
+    public String viewSchedule(Model model, OAuth2AuthenticationToken token, @PathVariable("scheduleid") Long scheduleid) {
+        
+        logger.info("Inside /schedule controller method ScheduleController#viewschedule");
+        logger.info("model=" + model + " token=" + token);
+
+        if (token!=null) {
+            String uid = token.getPrincipal().getAttributes().get("sub").toString();
+            logger.info("uid="+uid);
+            logger.info("scheduleItemRepository="+scheduleItemRepository);
+            List<Schedule> myschedules = scheduleRepository.findByUid(uid);// get all schedule ids by uid
+            // get courses by each scheduleid to a list
+            Iterable<ScheduleItem> myclasses = scheduleItemRepository.findByScheduleid(scheduleid);
+            Schedule schedule = new Schedule();
```

Comment:
Is there a reason for why a new schedule must be created at this step, if it will be replaced in the end?

Reply:
Agreed.